File: problemas/ad-hoc/3109/3109.r
input <- file('stdin', 'r')

n <- as.integer(readLines(input, n=1))

mesas <- c()
for(i in 1:n) {
    mesas <- c(mesas, i)
}

q <- as.integer(readLines(input, n=1))
for(i in 1:q) {
    entrada <- as.integer(strsplit(readLines(input, n = 1), ' ')[[1]])
    comando <- entrada[1]

    if (comando == 1) {
        a <- entrada[2]
        b <- entrada[3]

        temp <- mesas[a]
        mesas[a] <- mesas[b]
        mesas[b] <- temp
    } else {
        a <- entrada[2]

        resposta <- 0
        mesa <- mesas[a]
        while(mesa != a) {
            resposta <- resposta + 1
            mesa <- mesas[mesa]
        }

        write(paste(resposta), '')
    }
}

File: problemas/ad-hoc/3109/3109.py
N = int(input())
mesas = list(range(N + 1))

Q = int(input())
for _ in range(Q):
  query = [int(x) for x in input().strip().split(' ')]
  if (query[0] == 1):
    A, B = query[1:]
    mesas[A], mesas[B] = mesas[B], mesas[A]
  else:
    A = int(query[1])
    mesa = mesas[A]
    voltas = 0
    while mesa != A:
      voltas += 1
      mesa = mesas[mesa]
    print(voltas)
